Feat/cardano

AIQUANT7 opened the pull request:
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Adding a new chain for the Cardano Blockchain along with Minswap & Sundaeswap dex.
Allow users to trade and add/remove liquidity on AMM on these Cardano Dexs.


**Tests performed by the developer**:
Manual testing as well as Unit testing. 
path for test cases: test/chains/cardano
path for AMM tests Minswap: test/connectors/minswap
path for AMM tests Sundaeswap: test/connectors/sundaeswap

**Tips for QA testing**:

Can execute swap, add and remove liquidity in both Cardano Dex.

Aiquant1 committed:
Added cardano blockchain and it's route implementation

Aiquant1 committed:
Added minswap pool-info, quote-swap route

Aiquant1 committed:
Written swap route for minswap along with utility functions

Aiquant1 committed:
Written routes to executeSwap, AddLiquidity and removeLiquidity

Aiquant1 committed:
Written routes for executeSwap, AddLiquidity, and Remove Liquidity for sundaeswap also written test cases for cardano chain and minswap and sundaeswap

Aiquant1 committed:
Fixed remove_liquidity action for sundaeswap and remove unnecessary logs

Aiquant1 committed:
updated cardano.yml

Aiquant1 committed:
Fixed lucid instance

fengtality commented:
Thanks for the contribution!

If you resolve the conflicts (likely related to final v2.8 changes), I will review and test.

Aiquant1 committed:
fixed config routes for cardano and changed api and folder structure for cardano support and fixed tokens routes for cardano and added three network mainnet, preprod and preview for tokens

Aiquant1 committed:
refactor: implement lazy initialization for Cardano balance routes

- Move Cardano.getInstance() from route registration to request handlers
- Replace dynamic wallet address examples with static placeholders
- Prevent server startup crashes caused by premature blockchain connection
- Align Cardano initialization pattern with other chain implementations

fixed wallet, chains, pools route for cardano

Aiquant1 committed:
feat(cardano): add full AMM DEX support for Minswap and SundaeSwap

- Implement liquidity operations (add/remove/quote) for both DEXs
- Add pool info and swap quote endpoints
- Fix token address normalization for Cardano native assets
- Support multiple token address formats (dot notation and concatenated)
- Add Cardano wallet integration and validation utilities

fengtality commented:
Given that there's custom libs being imported, we need to verify that the DEXs and Cardano themselves are supportive of this connector. Otherwise, there's too many security risks for us to import these libs.

Please have the folks behind this PR reach out to me on the Hummingbot Discord or Twitter @fengtality.

AIQUANT7 commented:
> Given that there's custom libs being imported, we need to verify that the DEXs and Cardano themselves are supportive of this connector. Otherwise, there's too many security risks for us to import these libs.
> 
> ```
> "@aiquant/lucid-cardano": "^0.10.11",
>     "@aiquant/minswap-sdk": "^0.0.0",
>     "@aiquant/sundaeswap-core": "^1.2.10",
>     "@blockfrost/blockfrost-js": "^5.7.0",
> ```
> 
> Please have the folks behind this PR reach out to me on the Hummingbot Discord or Twitter @fengtality.

Blockfrost is not a custom lib it is a of chain node connector for Cardano, majorly used for on chain operations. As this is a dex we can interact directly with the dexs and make trades there, If the issue with the libs is there let's have a discission on discord regarding this thank you for your id, I will reach out. As we are developing the libs and keeping it up to date with core libs like sudeswap-core and minswap-sdk and lucid-cardano. Blockfrost is a Cardano blockchain library to connect to an remote node provided as a service, it is actively maintained by other developer in Cardano and is one of the core libraries in off chain to on chain connectors.

Aiquant1 committed:
fix(cardano): correct base/quote token handling for Minswap and SundaeSwap

- Fix base/quote token logic to match industry standards
- Fix asset mappings for any pool structure
- Fix amount parameter to use base token units
- Update all DEX routes (swap, liquidity, execute)
